src/main/java/school/lemon/changerequest/java/io/pr1/task3/CountEngineImp.java

```diff
@@ -0,0 +1,163 @@
+package school.lemon.changerequest.java.io.pr1.task3;
+
+import java.io.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Created by User on 03.02.2017.
+ */
+public class CountEngineImp implements CountEngine {
+
+    @Override
+    public long countSpecificCharacter(String filename, char c) throws FileNotFoundException {
+
+        long result = 0;
+        File file = new File(filename);
+        BufferedInputStream bis;
+        FileInputStream fis;
+        if (!file.exists()) throw new FileNotFoundException();
+        try {
+            fis = new FileInputStream(file);
+            bis = new BufferedInputStream(fis);
+            while (bis.available() > 0) {
+                if ((char) bis.read() == c) result++;
+            }
+        } catch (IOException n) {
+        }
+        return result;
+    }
+
+    @Override
+    public long countCharacters(String filename) throws FileNotFoundException {
+        long result = 0;
+        File file = new File(filename);
+        BufferedInputStream bis;
+        FileInputStream fis;
+        if (!file.exists()) throw new FileNotFoundException();
+        try {
+            fis = new FileInputStream(file);
+            bis = new BufferedInputStream(fis);
+            while (bis.available() > 0) {
+                bis.read();
+                result++;
+            }
+        } catch (IOException n) {
+        }
+        return result;
+    }
+
+    @Override
+    public long countWords(String filename) throws FileNotFoundException {
+        File file = new File(filename);
+        if (!file.exists()) {
+            throw new FileNotFoundException();
+        }
+        char c;
+        int words = 0;
+        int charForWords = 0;
+        int spaces = 0;
+        boolean wordWasCounted = false;
+        boolean firstWord = true;
+        Pattern charsP = Pattern.compile("[a-zA-Z0-9]");
+        Pattern spacesP = Pattern.compile("[\n\\s]");
+        Matcher charsM;
+        Matcher spacesM;
+        try {
+            BufferedInputStream bufferedInputStream = new BufferedInputStream(new FileInputStream(file));
```

Comment:
I'd prefer to read whole ling and count words in one line. Instead of parsing and analyzing each character separately.
It will be much more simpler.